CDN 2021 chapter (#2540)
HTTPArchive/almanac.httparchive.org: src/content/en/2021/cdn.md

3. Core Web Vitals
4. Cache Hit v/s Cache Miss performance etc.,

**CDN Adoption**

The contents in a web page can be divided into 3 parts, namely:

1. Base HTML Page (Ex: [www.example.com](www.example.com))
2. Embedded First Party Content (Ex: images.example.com, css.example.com etc.,)
3. 3rd Party Content (Ex: Google Analytics, Advertisements etc.,)

From their inception, CDNs have been the go-to solution for delivering embedded content like: images, stylesheets, javascripts, fonts, etc.  This kind of  content doesn't change frequently, making it a good candidate for caching on a CDN’s proxy servers.

With the evolution of CDN technology an expressway was set up on the internet for non-cacheable assets.  This means HTMLs and APIs can now be delivered reliably and faster, compared to the standard TCP connection.

<p id="gdcalert1" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image1.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert2">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>

> ![alt_text](images/image1.png "image_tooltip")

Reviewer: Follow the steps in the [Figures Guide](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Figures-Guide) to get the charts embedded in the markdown

Looking at third party content, there is negative growth for CDN adoption. Compared to [Web Almanac 2019 - CDN Chapter](https://almanac.httparchive.org/en/2019/cdn), we see 3% reduction in domains using CDNs. Third party domains are used by SaaS vendors for analytics, advertisements, responsive pages, etc. It is in the SaaS vendor’s interest to use CDNs for their services. Their content is used by multiple web owners and this content gets accessed by end users across geographies, making CDNs necessary from both a business and performance standpoint. This is evident in the charts where it's clear that third party content has the highest adoption of CDN.

But why do we see this negative growth in CDN Adoption for third party domains?

The probable reasons for this may include:

* The HTTP/2 protocol requires web owners to consolidate the domains instead of using multiple domains for optimal performance.

> * Contribution of third party content to total page weight has also increased over the years (refer to the (Third Parties)[./third-parties] chapter for more details) leading to increased page load time concerns for web owners.

Reviewer: ```suggestion
* Contribution of third party content to total page weight has also increased over the years (refer to the [Third Parties](./third-parties) chapter for more details) leading to increased page load time concerns for web owners.
```

Looking at third party domains supporting newer protocols, we see an interesting trend of higher adoption of HTTP/2+protocols compared to first party domains. This makes sense, considering the fact that most of the top third party domains use purpose-built CDNs and thus have more control on the content development and content delivery. Additionally, third party domains need to have consistent performance across all network conditions and this is where HTTP/2+ adds value by mixing in other protocols like UDP along with TCP connections. Back in 2019, Uber did an experiment to understand how UDP along with TCP (aka QUIC, the precursor to HTTP/3) can help deliver content with consistent performance and overcome packet loss in highly congested mobile networks. The results of this experiment documented in [this blog post](https://eng.uber.com/employing-quic-protocol/) throws valuable insights into the demographic where HTTP/3 can help. Over time, this trend will trickle down and we should see web owners adopting HTTP/3, especially with mobile network traffic having a higher contribution to the total internet traffic.

**Brotli Adoption**

> Content delivered over the internet employs compression to reduce the payload size. A smaller payload means it’s faster to deliver the content from server to end user. This makes websites load faster and provide a better end user experience. For images, this compression is handled by image file formats like JPEG, WEBP, AVIF, etc. (refer to (Media)[./media] chapter for more on this). For textual web assets (like HTML, Javascripts, stylesheets, etc.) compression is handled by a file format called “[GZIP](https://en.wikipedia.org/wiki/Gzip)”. GZIP has been in existence since 1992.  It did a good job of making text asset payloads smaller,but it is time for a new text asset compression to do better than GZIP:[Brotli](https://en.wikipedia.org/wiki/Brotli) (refer to (Compression)[./compression] chapter for more on this). 

Reviewer: ```suggestion
Content delivered over the internet employs compression to reduce the payload size. A smaller payload means it’s faster to deliver the content from server to end user. This makes websites load faster and provide a better end user experience. For images, this compression is handled by image file formats like JPEG, WEBP, AVIF, etc. (refer to [Media](./media) chapter for more on this). For textual web assets (like HTML, Javascripts, stylesheets, etc.) compression is handled by a file format called “[GZIP](https://en.wikipedia.org/wiki/Gzip)”. GZIP has been in existence since 1992.  It did a good job of making text asset payloads smaller,but it is time for a new text asset compression to do better than GZIP:[Brotli](https://en.wikipedia.org/wiki/Brotli) (refer to the [Compression](./compression) chapter for more on this). 
```

> featured_stat_label_3: TODO

Reviewer: Could you add a bio for yourself and also include featured quote/stats? See the [wiki](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide#metadata-to-add-at-the-top-of-your-chapters) for more info.

Please also add `unedited: true`
```suggestion
featured_stat_label_3: TODO
unedited: true
```

With this in mind, here are a few statistics that were intentionally not measured in the context of a CDN:

1. Time To First Byte(TTFB)
2. CDN Round Trip Time
3. Core Web Vitals
4. Cache Hit v/s Cache Miss performance etc.,

**CDN Adoption**

The contents in a web page can be divided into 3 parts, namely:

1. Base HTML Page (Ex: [www.example.com](www.example.com))
2. Embedded First Party Content (Ex: images.example.com, css.example.com etc.,)
3. 3rd Party Content (Ex: Google Analytics, Advertisements etc.,)

From their inception, CDNs have been the go-to solution for delivering embedded content like: images, stylesheets, javascripts, fonts, etc.  This kind of  content doesn't change frequently, making it a good candidate for caching on a CDN’s proxy servers.

With the evolution of CDN technology an expressway was set up on the internet for non-cacheable assets.  This means HTMLs and APIs can now be delivered reliably and faster, compared to the standard TCP connection.

{{ figure_markup(
  image="CDN Usage vs Hosted Resources.png",
  caption="CDN Usage vs Hosted Resources.",

>   description="Bar chart showing the CDN Usage for domains serving HTML, sub-domains serving static assets, third party domains",

Reviewer: It'd be helpful to be more descriptive in all figures `description` fields. If you were to try to describe the contents of the figure to someone over the phone, what would you say?

For example, in this chart I'd say something like this:
```suggestion
  description="Bar chart of CDN usage versus hosted resource split by origin and CDN. For HTML content, 73% of requests are from the origin and 27% from CDNs. For sub-domain content, 62% origin and 38% CDN. And for third-party content, 37% origin and 63% CDN.",
```
